subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/api/types.go

```diff
@@ -158,3 +158,51 @@ type LazyPreemptionStatus struct {
 	// It was lazy preempted at PreemptionTime.
 	PreemptionTime meta.Time `json:"preemptionTime"`
 }
+
+type ClusterStatus struct {
+	// Status of cells in the physical cluster
+	PhysicalCluster []*PhysicalCellStatus `json:"physicalCluster"`
+	// Status of cells in each VC
+	VirtualClusters map[string][]*VirtualCellStatus `json:"virtualClusters"`
+}
+
+type CellStatus struct {
+	GpuType  string `json:"gpuType,omitempty"`
+	CellType string `json:"cellType"`
+	// Address of a physical cell consists of its address (or index) in each level
+	// (e.g., node0/0/0/0 may represent node0, CPU socket 0, PCIe switch 0, GPU 0.
+	// Address of a virtual cell consists of its VC name, index of the preassigned cell,
+	// and the relative index in each level inside the preassigned cell
+	// (e.g., VC1/0/0 may represent VC1, preassigned cell 0, index 0 among its children)
+	CellAddress string `json:"cellAddress"`
+	// State can be one of "Free", "Used", and "Bad"
+	State    string `json:"state"`
+	Priority int32  `json:"priority"`
```

Comment:
Using Enum, like

// [AllocatedState]: The Pod is considered to be allocated from the scheduler view.
const (
 // Pod is waiting for preemptible or free resource to appear.
 // [StartState]
 // -> PodPreempting
 // -> PodBinding
 PodWaiting PodState = "Waiting"

 // Pod is waiting for the appeared preemptible resource to be free by preemption.
 // -> PodBinding
 // -> PodWaiting
 PodPreempting PodState = "Preempting"
}